src/shogun/lib/SGMatrix.h

```diff
@@ -309,6 +333,68 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** fill matrix with zeros */
 		void zero();
 
+		/**
+		 * fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		 * Rest: disabled
+		**/
+		// vectorized available
+		template <typename U>
+		using enable_simd_float = std::enable_if_t<is_any<U, float64_t>::value>;
+		template <typename U = T>
+		auto random() -> enable_simd_float<U>
+		{
+			CRandom r{};
+			r.fill_array_co(
+			    static_cast<float64_t*>(matrix), num_rows * num_cols);
```

Comment:
Here you could write `static_cast<U*>(matrix), num_rows * num_cols)` instead, no?

Reply:
Yep I could. Will do so!

Yea I wanted to get the Thumbs up on this version. Adding it for `SGVector` shouldnt take too long on top of this. Dont think there's any other container right?

Also the type trait `is_any` <- I think this should go in some other file where it can be reused. Maybe `common.h`?

Reply:
I think so, you will need @karlnapf to get back to you on that one though. Btw I think you will need to rename `is_any`, because there is an Any class in shogun, which might cause some confusion. Is `is_any_of` too verbose? 
Btw if you always use the value of `is_any`, instead of its type, you could write a `is_any_v` as a short cut to `is_any<...>::value`. I think that is more conforming with the direction in which c++ is going. (see for example [std::is_void_v](https://en.cppreference.com/w/cpp/types/is_void))

Reply:
+1 for the rename...
Actually, we might even go explicit for now as as there are not THAT many combinations...I like it on the other hand, so could also keep it as a generic tool within the lib (under a different name)

Reply:
done